### PR TITLE
Make shard accessible to history handler

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Include current shard time to APIs below
  * StartWorkflowExecutionResponse
  * RecordWorkflowTaskStartedResponse
  * RecordActivityTaskStartedResponse

**Why?**
see #2743 

**How did you test it?**
N/A

**Potential risks**
N/A

**Is hotfix candidate?**
N/A

### CODE REVIEW

File: service/history/handler.go [hunk 1]
```diff
@@ -247,9 +247,13 @@ func (h *Handler) RecordActivityTaskHeartbeat(ctx context.Context, request *hist
...
-	engine, err1 := h.controller.GetEngine(ctx, namespaceID, workflowID)
-	if err1 != nil {
-		return nil, h.convertError(err1)
+	shardContext, err := h.controller.GetShardByNamespaceWorkflow(ctx, namespaceID, workflowID)
+	if err != nil {
+		return nil, h.convertError(err)
+	}
+	engine, err := shardContext.GetEngineWithContext(ctx)
+	if err != nil {
+		return nil, h.convertError(err)
```
Comment (yiminc) [inline service/history/handler.go:256]: Why doing this all over the places? This belong to GetEngine method.

Comment (wxing1292) [reply service/history/handler.go:256]: https://github.com/temporalio/temporal/pull/2745/files#diff-45c4598dbf388574168a77db0f3c6c2dc52aa3f5a34bd9edbb6e796b10b9de9fR292

feel free to ask me to create a new API, i.e. `GenerateVClock()`

Comment (yux0) [reply service/history/handler.go:256]: Can this be done in history engine?

Comment (wxing1292) [reply service/history/handler.go:256]: of course, but my long term goal is to make history engine go away:
* API logic should be moved out, e.g. one API per package
* @yycptt will move the per shard / engine task processing logic out

**Review by yycptt — APPROVED** [review]

**Pull request merged**